app/screens/home/search/modifiers/index.tsx

```diff
@@ -0,0 +1,157 @@
+        },
+        separatorContainer: {
+            justifyContent: 'center',
+            flex: 1,
+            height: RECENT_SEPARATOR_HEIGHT,
+        },
+        separator: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+            height: 1,
+        },
+        sectionList: {
+            flex: 1,
+        },
+    };
+});
+
+const getModifiersSectionsData = (intl: IntlShape): ModifierItem[] => {
+    const formatMessage = intl.formatMessage;
+    const sectionsData = [
+        {
+            term: 'From:',
+            testID: 'search.from_section',
+            description: formatMessage({id: 'mobile.search.modifier.from', defaultMessage: ' a specific user'}),
+        }, {
+            term: 'In:',
+            testID: 'search.in_section',
+            description: formatMessage({id: 'mobile.search.modifier.in', defaultMessage: ' a specific channel'}),
+        }, {
+            term: 'On:',
+            testID: 'search.on_section',
+            description: formatMessage({id: 'mobile.search.modifier.on', defaultMessage: ' a specific date'}),
+        }, {
+            term: 'After:',
+            testID: 'search.after_section',
+            description: formatMessage({id: 'mobile.search.modifier.after', defaultMessage: ' after a date'}),
+        }, {
+            term: 'Before:',
+            testID: 'search.before_section',
+            description: formatMessage({id: 'mobile.search.modifier.before', defaultMessage: ' before a date'}),
+        }, {
+            term: '-',
+            testID: 'search.exclude_section',
+            description: formatMessage({id: 'mobile.search.modifier.exclude', defaultMessage: ' exclude search terms'}),
+        }, {
+            term: '""',
+            testID: 'search.phrases_section',
+            description: formatMessage({id: 'mobile.search.modifier.phrases', defaultMessage: ' messages with phrases'}),
+        },
+    ];
+    return sectionsData;
+};
+
+type Props = {
+    scrollPaddingTop: number;
+    setSearchValue: (value: string) => void;
+    searchValue?: string;
+}
+const SearchModifiers = ({scrollPaddingTop, searchValue, setSearchValue}: Props) => {
+    const theme = useTheme();
+    const intl = useIntl();
+    const paddingTop = useMemo(() => ({paddingTop: scrollPaddingTop, flexGrow: 1}), [scrollPaddingTop]);
+
+    const [showMore, setShowMore] = useState(false);
+    const show = useSharedValue(3 * MODIFIER_LABEL_HEIGHT);
+    const data = useMemo(() => getModifiersSectionsData(intl), [intl]);
+
+    const styles = getStyleFromTheme(theme);
+    const animatedStyle = useAnimatedStyle(() => (
+        {
+            width: '100%',
+            height: withTiming(show.value, {duration: 300}),
+            overflow: 'hidden',
+        }
+    ));
+
+    const handleShowMore = useCallback(() => {
+        setShowMore(!showMore);
+        show.value = (showMore ? 3 : data.length) * MODIFIER_LABEL_HEIGHT;
```

Comment:
(0/5) This is a bit confusing. For readability sake I would go with something like:
```suggestion
        const nextShowMore = !showMore;
        show.value = (nextShowMore ? data.length : 3) * MODIFIER_LABEL_HEIGHT;
        setShowMore(nextShowMore);
```